lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -797,7 +797,7 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 		// Post slug.
-		if ( ! empty( $schema['properties']['slug'] ) && isset( $request['slug'] ) ) {
+		if ( ! empty( $schema['properties']['slug'] ) && isset( $request['slug'] ) && ! empty( $request['slug'] ) && ! is_object( $request['slug'] ) ) {
```

Comment:
@rmccue @joehoyle I tracked this down to our use of the `sanitize_title` to sanitize the slugs. `sanitize_title` takes an optional second parameter as a 'fallback'. in class-wp-rest-controller:L807 in `sanitize_params` where we use `call_user_func`, we are passing the value, then also passing the request object and the argument key.

In this case `sanitize_title` sees an empty string as failing and returns the fallback, the request object, and the post update fails.

Do you know why the request and key are added to the sanitization callback? can those be removed? Going to dig into when they were added.

https://core.trac.wordpress.org/browser/trunk/src/wp-includes/rest-api/class-wp-rest-request.php#L807